src/core/qgis.h

```diff
@@ -346,6 +346,20 @@ class CORE_EXPORT Qgis
     };
     Q_ENUM( PythonMacroMode )
 
+    /**
+     * Authorisation to load Python Expression Functions from project
+     * \since QGIS 3.40
+     */
+    enum class PythonExpressionFromProjectMode SIP_MONKEYPATCH_SCOPEENUM_UNNEST( Qgis, PythonExpressionFromProjectMode ) : int
```

Comment:
```suggestion
    enum class PythonExpressionFromProjectMode : int
```

There's no need for the compatibility code for new enums.

That said -- I'd lean toward just making the current macro setting apply to expressions too, and reword the labels / options around it to indicate that it applies to embedded Python code like macros or user expressions instead.

Reply:
Thanks for the review! Now we're using only the macro setting to know if we should load both macros and expression functions from projects.

Should we rename the enum itself? Instead of `PythonMacroMode` use something like `EmbeddedPythonMode`?